src/Microsoft.IdentityModel.Tokens/Validation/Results/ValidatedSigningKeyLifetime.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+#nullable enable
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Represents a validated signing key lifetime.
+    /// </summary>
+    internal readonly struct ValidatedSigningKeyLifetime : IEquatable<ValidatedSigningKeyLifetime>
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="ValidatedSigningKeyLifetime"/>.
+        /// </summary>
+        /// <param name="validFrom">The date from which the signing key is considered valid.</param>
+        /// <param name="validTo">The date until which the signing key is considered valid.</param>
+        /// <param name="validationTime">The time the validation occurred.</param>
+        internal ValidatedSigningKeyLifetime(DateTime? validFrom, DateTime? validTo, DateTime? validationTime)
+        {
+            ValidFrom = validFrom;
+            ValidTo = validTo;
+            ValidationTime = validationTime;
+        }
+
+        /// <summary>
+        /// The date from which the signing key is considered valid.
+        /// </summary>
+        public DateTime? ValidFrom { get; }
+
+        /// <summary>
+        /// The date until which the signing key is considered valid.
+        /// </summary>
+        public DateTime? ValidTo { get; }
+
+        /// <summary>
+        /// The time the validation occurred.
+        /// </summary>
+        public DateTime? ValidationTime { get; }
+
```

Comment:
Can we add a ToString() method that displays `$"{ValidationTime} ∊ [{ValidFrom}, {Validto}]"`